### PR TITLE
Cache pin persistent

### PR DESCRIPTION
**Pull request by JulianKoch**
Ready for review.

### CODE REVIEW

**Comment by alexfupp** [comment]
I did a very simple basic test: save a pin, restart the app and try to use the pin. This currently leads to FormatException. Please fix this first.

While having a very quick view on the code I noticed that the cached pin value is quite often converted between String and Int. Try to mostly stick with only one format and convert as little as possible.

File: de.persosim.driver.connector.ui/src/de/persosim/driver/connector/ui/parts/ReaderPart.java [hunk 1]
```diff
@@ -381,51 +393,124 @@ public void widgetDefaultSelected(SelectionEvent e) {
...
 		return button;
 
 	}
 	
+	/**returns true if a key e.g. b1 exists in the preferences
+	 * 
+	 * @param the key to check
+	 * @return true/false
+	 */
+	public boolean checkKey(String key) {
```
Comment (alexfupp) [inline de.persosim.driver.connector.ui/src/de/persosim/driver/connector/ui/parts/ReaderPart.java]: The name of this method is quite bad. It does not reflect what is checked...

**Comment by JulianKoch** [comment]
After testing on different machines: The FormatException should no longer appear. It was thrown by the split() method which delivers different results in Java 7 and 8. The number of unnecessary conversations between String and int for pin is reduced and the checkKey() got an new name & better description.

**Comment by alexfupp** [comment]
Haven't looked at the code at all, but I still experience strange behaviour when testing:
- Reset seems to scramble the button numbers
- Still get a NumberFormatException when recalling a previously saved PIN

**Comment by alexfupp** [comment]
sorry, I mixed up some versions. The NumberFormatException seems fixed.
The issue regarding button numbering persists.

**Comment by alexfupp** [comment]
added some comments/tasks in alexfupp@3098bed97dec205ae786e998c10be2705ba7cbe3

**Comment by JulianKoch** [comment]
Should be ok now.

**Pull request merged**